compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -202,13 +204,14 @@ FailureOr<ParameterSetting> ParameterSetting::create(
     uint32_t kPackScaleL1) {
   auto initType =
       llvm::cast<ShapedType>(linalgOp.getDpsInitOperand(0)->get().getType());
-  unsigned nBytesInit = initType.getElementTypeBitWidth() / 8;
+  uint32_t nBytesInit = initType.getElementTypeBitWidth() / 8;
+  uint32_t nBytesOut = nBytesInit;
```

Comment:
I'm not clear why this change is made. I think previously `nBytesInit` is for matmul output, and `nBytesElemOut` is not zero only when there are elementwise ops. If you use both `nBytesInit` and `nBytesOut` for pure matmul, it will count extra memory when calculating L1 memory usage?

Reply:
Thanks for catching that! I think we can solve it by initializing `nBytesOut` to `0`, so I updated it.

Reply:
Actually, there was no double count, so I reverted it back as I need `nBytesOut` to compute L2 usage.

Reply:
Why there's no double count? The calculation is as
```
uint32_t A = params.numBytesA * m * k * params.bufferDepthA;
uint32_t B = params.numBytesB * n * k * params.bufferDepthB;
uint32_t C = params.numBytesC * m * n * params.bufferDepthC;
uint32_t Acc = params.numBytesAcc * m * n * params.bufferDepthAcc;
int64_t memoryUsage = A + B + C + Acc;
```
If `params.bufferDepthC` is not zero for matmul, I think it will count more memory usage.

Reply:
I know you need `nBytesOut` for L2 memory calculation.  I think you can keep `nBytesElemOut` while adding a new variable as `nBytesOut` after L1 memory usage calculation as
`uint32_t nBytesOut = isMatmulWithElementwiseConsumer(linalgOp) ? nBytesElemOut : nBytesInit`.

Reply:
Oh, I see what you mean. Yeah, there is some potential double count. I updated it to keep nBytesElemOut and pass that for L1 memory calculation.